### PR TITLE
Use a pool connection

### PR DESCRIPTION
**Pull request by Sytten**
I tried not to break the API since we have a V1 published.
That was quite bad before, if the server dropped the connection it would not be able to reconnect later...

### CODE REVIEW

**Comment by francistb** [comment]
looks good !

**Comment by malaporte** [comment]
Looks fine here too!

File: src/main/java/com/coveo/spillway/storage/RedisStorage.java [hunk 1]
```diff
@@ -55,57 +58,144 @@
...
+      String prefix,
+      int timeout,
+      String password,
+      int database) {
+    jedisPool = new JedisPool(poolConfig, host, port, timeout, password, database);
+    keyPrefix = prefix;
+  }
+
+  /**
+   * Will be remove in V1.2.0
```
Comment (GuiSim) [inline src/main/java/com/coveo/spillway/storage/RedisStorage.java]: removed :)

File: src/main/java/com/coveo/spillway/storage/RedisStorage.java [hunk 2]
```diff
@@ -55,57 +58,144 @@
...
+   *
+   * @param jedis A Jedis connection
+   */
+  @Deprecated
   public RedisStorage(Jedis jedis) {
```
Comment (GuiSim) [inline src/main/java/com/coveo/spillway/storage/RedisStorage.java]: Worth it to replace with a constructor that accepts a `JedisPool`?

File: src/main/java/com/coveo/spillway/storage/RedisStorage.java (hunk 1 above)
Comment (jebeaudet) [inline src/main/java/com/coveo/spillway/storage/RedisStorage.java]: removed

You can't remove a constructor unless you bump the major version. Don't mention it here anyway

File: src/main/java/com/coveo/spillway/storage/RedisStorage.java (hunk 2 above)
Comment (GuiSim) [inline src/main/java/com/coveo/spillway/storage/RedisStorage.java]: Too bad, it's a basic concept in dependency injection and very useful to ensure maximum flexibility.

For example if a client needs to set something that we haven't exposed, we don't have to add a new constructor. We can simply accept the JedisPool directly. e.g. if someone needs to set something like proxy settings or whatever setting that we do not care about.

**Pull request merged**